src/gazebo_geotagged_images_plugin.cpp

```diff
@@ -31,346 +31,585 @@ using namespace std;
+    static const char* model  = "Gazebo";
+    char uri[128] = {};
+    uint32_t camera_capabilities = CAMERA_CAP_FLAGS_CAPTURE_IMAGE;
+    mavlink_message_t msg;
+    mavlink_msg_camera_information_pack_chan(
+        1,
+        MAV_COMP_ID_CAMERA,
+        MAVLINK_COMM_1,
+        &msg,
+        0,                         // time_boot_ms
+        (const uint8_t *)vendor,   // const uint8_t * vendor_name
+        (const uint8_t *)model,    // const uint8_t * model_name
+        0x01,                      // uint32_t firmware_version
+        50.0f,                     // float focal_lenth
+        35.0f,                     // float  sensor_size_h
+        24.0f,                     // float  sensor_size_v
+        _width,                    // resolution_h
+        _height,                   // resolution_v
+        0,                         // lens_id
+        camera_capabilities,       // CAP_FLAGS
+        0,                         // Camera Definition Version
+        uri                        // URI
+    );
+    _send_mavlink_message(&msg, srcaddr);
+}
 
-  if (destination_port != 0) {
-    _srcaddr.sin_port = htons(destination_port);
-  }
+void GeotaggedImagesPlugin::_handle_request_camera_settings(const mavlink_message_t *pMsg, struct sockaddr* srcaddr)
+{
+    gzdbg << "Send camera settings" << endl;
+    mavlink_command_long_t cmd;
+    mavlink_msg_command_long_decode(pMsg, &cmd);
+    //-- Should we execute the command
+    if ((int)cmd.param1 != 1)
+    {
+        gzwarn << "Request ignored" << endl;
+        return;
+    }
+    // ACK command received and accepted
+    _send_cmd_ack(pMsg->sysid, pMsg->compid, MAV_CMD_REQUEST_CAMERA_SETTINGS, MAV_RESULT_ACCEPTED, srcaddr);
+    mavlink_message_t msg;
+    mavlink_msg_camera_settings_pack_chan(
+        1,
+        MAV_COMP_ID_CAMERA,
+        MAVLINK_COMM_1,
+        &msg,
+        0,                      // time_boot_ms
+        CAMERA_MODE_IMAGE);     // Camera Mode
+    _send_mavlink_message(&msg, srcaddr);
+}
 
-  ssize_t len = sendto(_fd, buffer, packetlen, 0, (struct sockaddr *)&_srcaddr, sizeof(_srcaddr));
+void GeotaggedImagesPlugin::_send_capture_status(struct sockaddr* srcaddr)
+{
+    _captureMutex.lock();
+    int status = _captureMode == CAPTURE_DISABLED ? 0 : (_captureMode == CAPTURE_SINGLE ? 1 : 3);
+    _captureMutex.unlock();
+    gzdbg << "Send capture status" << endl;
+    float available_mib = 0.0f;
+    boost::filesystem::space_info si = boost::filesystem::space(".");
+    available_mib = (float)((double)si.available / (1024.0 * 1024.0));
+    mavlink_message_t msg;
+    mavlink_msg_camera_capture_status_pack_chan(
+        1,
+        MAV_COMP_ID_CAMERA,
+        MAVLINK_COMM_1,
+        &msg,
+        0,
+        status,                                 // image status
+        0,                                      // video status (Idle)
+        _captureMode == CAPTURE_ELAPSED ? (float)_captureInterval : 0, // image interval
```

Comment:
`_captureMode` is not guarded by `_captureMutex` here. Is this on purpose?

Reply:
Nope, I missed it.